app/questionnaire/questionnaire_store_updater.py

```diff
@@ -526,19 +473,23 @@ def remove_dependent_blocks_and_capture_dependent_sections(self) -> None:
                 )
                 blocks_removed |= self.remove_completed_location(location)
 
-            if blocks_removed and (
-                section_id != self._current_location.section_id
-                or list_item_id != self._current_location.list_item_id
-            ):
-                # Since this section key will be marked as incomplete, any `DependentSection` with is_complete as `None`
-                # can be removed as we do not need to re-evaluate progress as we already know the section would be incomplete.
-                dependent = DependentSection(section_id, list_item_id)
-                if dependent in self.dependent_sections:
-                    self.dependent_sections.remove(dependent)
-
-                self.dependent_sections.add(
-                    DependentSection(section_id, list_item_id, is_complete=False)
-                )
+            if blocks_removed:
+                self._capture_dependent_section(section_id, list_item_id)
```

Comment:
Minor, and optional, could use `section_key`. Reduces the number of args, prevent passing incorrect order and section key only have attributes that are needed and no bloat.

Reply:
done 👍